core/src/main/java/com/microsoft/applicationinsights/internal/perfcounter/JniPCConnector.java

```diff
@@ -126,15 +140,94 @@ private static boolean initNativeCode() {
         return true;
     }
 
-    private static boolean loadLibrary() {
+    /**
+     * The method will try to extract the dll for the Windows performance counters to a local
+     * folder and then will try to load it. The method will do all that by doing the following things:
+     * 1. Find the OS type (64/32) currently supports only 64 bit.
+     * 2. Will find the path to extract to, which is %temp%/sdk_version_number
```

Comment:
The path should include at least another level for grouping applicationinsights data, i.e. %temp%/<applicationinsights>/<sdk_version_number>, or even better %temp%/<applicationinsights>/<artifact_name>/<sdk_version_number>

This will also be a good preparation if we will decide to change the location under which we persist throttled telemetry.